### PR TITLE
Allow ci-admins to crud prowjobs

### PR DESCRIPTION
**Pull request by 0xmichalis**
/cc @enj

### CODE REVIEW

**Comment by enj** [comment]
Does this need to be compatible with older clusters?  If not, define the new role / bindings using RBAC instead of origin authz.  Also, I believe @smarterclayton said he wanted ci-admins to be admins cluster wide.  But maybe he wanted a different group with that power instead?

File: cluster/ci/config/roles.yaml [hunk 1]
```diff
@@ -30,6 +24,8 @@ items:
   - stevekuznetsov
   - jwforres
   - kargakis
+  - eparis
+  - enj
```
Comment (stevekuznetsov) [inline cluster/ci/config/roles.yaml]: ?

**Comment by kargakis** [comment]
> define the new role / bindings using RBAC instead of origin authz

Unclear what this means.. Do you mean that I should use the upstream types?

**Comment by stevekuznetsov** [comment]
> wanted ci-admins to be admins cluster wide. But maybe he wanted a different group with that power instead?

@enj we have `cluster-reader` and `sudoer` which is sufficient. Giving access to _e.g._ `oc delete` is dangerous (lol)

File: cluster/ci/config/roles.yaml (hunk 1 above)
Comment (kargakis) [reply cluster/ci/config/roles.yaml]: @enj is already in the ci-admins group in our cluster. I am not sure what's the scope of the group honestly. cc: @smarterclayton

Comment (stevekuznetsov) [reply cluster/ci/config/roles.yaml]: This group gives cluster level access to everything. I don't think Mo needs that.

Comment (enj) [reply cluster/ci/config/roles.yaml]: I was given access to help figure out some OAuth / RBAC issues.  You are welcome to scope that down as you see fit.

**Comment by enj** [comment]
> Unclear what this means.. Do you mean that I should use the upstream types?

Yes, there are the canonical representation in 3.7.  The Origin authz resources are a proxy.

> we have cluster-reader and sudoer which is sufficient. Giving access to e.g. oc delete is dangerous (lol)

That is fine, but it will fail the SAR check that Jenkins performs.

**Comment by stevekuznetsov** [comment]
We should have `admin` in the `ci` namespace from our GitHub team sync, that should take care of Jenkins

**Comment by kargakis** [comment]
Keeping just the cluster{role,rolebinding} for now. @smarterclayton where do I need to plumb this manifest so it is applied on cluster creation/upgrade?

**Comment by enj** [comment]
@kargakis let the server do the conversion for you:

```
oc create -f prowjob_access.yaml
oc export clusterrolebinding.rbac.authorization.k8s.io/ci-admins-prowjobs clusterrole.rbac.authorization.k8s.io/prowjob-access -o yaml
```

```yaml
apiVersion: v1
items:
- apiVersion: rbac.authorization.k8s.io/v1beta1
  kind: ClusterRoleBinding
  metadata:
    creationTimestamp: null
    name: ci-admins-prowjobs
  roleRef:
    apiGroup: rbac.authorization.k8s.io
    kind: ClusterRole
    name: prowjob-access
  subjects:
  - apiGroup: rbac.authorization.k8s.io
    kind: Group
    name: ci-admins
- apiVersion: rbac.authorization.k8s.io/v1beta1
  kind: ClusterRole
  metadata:
    creationTimestamp: null
    name: prowjob-access
  rules:
  - apiGroups:
    - prow.k8s.io
    resources:
    - prowjobs
    verbs:
    - create
    - delete
    - get
    - list
kind: List
metadata: {}
```

**Comment by kargakis** [comment]
@enj I am not as worried. There are other roles inside this repo that need to be converted, too, and I would prefer to handle them all holistically.

**Comment by kargakis** [comment]
@smarterclayton we need to figure out how to apply this as part of the upgrade.

**Comment by kargakis** [comment]
or fresh installation

**Comment by smarterclayton** [comment]
Makefile in ci cluster was what i planned to run.

**Comment by kargakis** [comment]
Added in https://github.com/openshift/release/pull/206

**Pull request merged**